main/main.go

```diff
@@ -105,16 +104,23 @@ func main() {
 		return
 	}
 
-	// Create a health check server that polls a single atomic bool, settable by any relayer goroutine on failure
-	healthy := atomic.Bool{}
-	healthy.Store(true)
+	// Each goroutine will have an atomic bool that it can set to false if it ever disconnects from its subscription.
+	relayerHealth := make(map[ids.ID]*atomic.Bool)
```

Comment:
Oh never mind, I think I misunderstood how it's used. The values in the map (pointers) are only set once by the main goroutine. The other goroutines do not update (or have reference to) the map at all, only the pointer to their respective atomic bool. 👍

Reply:
We never delete items from the map and adding items to the map only happens at initialization. I can move the addition of keys above the creation of the health endpoint so that the health endpoint always sees the same set of keys.

As for setting the bools, they already need to happen in a linear codepath, or else one path could set the status to healthy after both codepaths set it to unhealthy (aka we need an independent bool for any reason the relayer could be unhealthy). Synchronization isn't going to help this.

The sync package map doesn't give the functionality we need here (there is a range funciton, but is more of a map-reduce) so it's not easy to swap it out.

We could make our own struct with a mutex and the desired functionality, but I really think it's unnecessary

Reply:
Yeah, I misunderstood and thought multiple routines would be updating the value in the map, but that's not the case.